spring-mvc/src/main/java/com/kodemon/springmvc/controllers/UserController.java

```diff
@@ -0,0 +1,93 @@
+package com.kodemon.springmvc.controllers;
+
+import com.kodemon.api.dto.UserAuthDTO;
+import com.kodemon.api.dto.UserDTO;
+import com.kodemon.api.facade.UserFacade;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import javax.inject.Inject;
+import javax.servlet.ServletRequest;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+import java.util.Collection;
+
+/**
+ *  @author Oliver Roch
+ */
+
+@Controller
+@RequestMapping("/user")
+public class UserController {
+
+    @Inject
+    private UserFacade userFacade;
+
+    /**
+     * Show list of all trainers.
+     *
+     * @param model data to display
+     * @return JSP page name
+     */
+    @RequestMapping(value = "/list", method = RequestMethod.GET)
+    public String list(Model model) {
+        model.addAttribute("users", userFacade.findAllUsers());
+        return "user/list";
+    }
+
+    /**
+     * Show detail of trainer specified by his username
+     *
+     * @param username which trainer is to be shown
+     * @param model data to display
+     * @return JSP page name
+     */
+    @RequestMapping(value = "/detail/{username}", method = RequestMethod.GET)
+    public String detail(@PathVariable String username, Model model) {
+        model.addAttribute("trainer", userFacade.findUserByUserName(username).iterator().next());
```

Comment:
Or something like you put in the `find` method, that seems acceptable.
When posting screenshots, please post one with that page/error displayed (No trainer found)